src/tiramisu_function.cpp

```diff
@@ -116,6 +116,103 @@ isl_union_map *tiramisu::function::compute_dep_graph() {
     return result;
 }
 
+const std::map<std::string, tiramisu::buffer *> tiramisu::function::get_mapping() const
+{
+  return this->mapping;
+}
+
+void  tiramisu::function::add_mapping(std::pair<std::string,tiramisu::buffer *> p)
+{
+  this->mapping.insert(p);
+}
+
+/**
+ * This function takes computation pts to C1 and C2 which are the first and the last computation
+ * of the fucntion.
+ * Automatic_communication fonction verifies if the user wants to manage the copies manually or not.
+ * By default, the data management will be done automatically, and the copies will be inserted
+ * in the begining and at the end of the gpu funcion.
```

Comment:
Explain at high level how you will decide whether you copy or not? (i.e., explain the algorithm at high level). This is where the algorithm should be documented.